infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batch/ReadBatch.java

```diff
@@ -0,0 +1,73 @@
+package org.corfudb.infrastructure.log.statetransfer.batch;
+
+import com.google.common.collect.ImmutableList;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+import org.corfudb.protocols.wireprotocol.IMetadata;
+import org.corfudb.protocols.wireprotocol.LogData;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static lombok.Builder.Default;
+
+/**
+ * A batch of data read from the cluster (or from a specific node) and
+ * an optional destination server where the corresponding data is located.
+ */
+@Getter
+@ToString
+@Builder
+public class ReadBatch {
+    public enum FailureStatus {
```

Comment:
Same, name `FailureStatus` is confusing, consider changing a name?

Reply:
Sure